src/sql/workbench/api/browser/mainThreadExtensionManagement.ts

```diff
@@ -29,4 +35,28 @@ export class MainThreadExtensionManagement implements MainThreadExtensionManagem
 	public $install(vsixPath: string): Thenable<string> {
 		return this._extensionService.install(URI.file(vsixPath)).then((value: ILocalExtension) => { return undefined; }, (reason: any) => { return reason ? reason.toString() : undefined; });
 	}
+
+	public $showObsoleteExtensionApiUsageNotification(message: string): void {
+		console.warn(message);
+
+		if (this._obsoleteExtensionApiUsageNotificationShown) {
+			return;
+		}
+
+		let enableObsoleteAPINotification = this._configurationService.getValue('workbench')['enableObsoleteApiUsageNotification'];
+		if (enableObsoleteAPINotification !== undefined && !enableObsoleteAPINotification) {
+			return;
+		}
+
+		this._notificationService.prompt(Severity.Warning,
+			localize('workbench.generalObsoleteApiNotification', "Some of the loaded extensions are using obsolete APIs, please find the detailed information in the console of Developer Tools"),
```

Comment:
Hm, I feel like this reads a little oddly. Maybe something like 

```
"Some loaded extensions are using obsolete APIs, please find detailed information in the Developer Tools Console"
```

I'd also suggest making Developer Tools Console into a command-link to open the developer tools